internal/widget/menu_item.go

```diff
@@ -100,44 +107,42 @@ func (i *MenuItem) Tapped(*fyne.PointEvent) {
 	i.Parent.dismiss()
 }
 
-func (i *MenuItem) activateChild() {
-	if i.Child != nil && i.Child.activeChild != nil {
-		i.Child.activeChild.Hide()
-		i.Child.activeChild = nil
-	}
-
-	if i.Parent.activeChild == i.Child {
-		return
-	}
-
-	if i.Parent.activeChild != nil {
-		i.Parent.activeChild.Hide()
-	}
-	i.Parent.activeChild = i.Child
-	if i.Child != nil {
-		i.Child.Show()
-	}
-}
-
-func (i *MenuItem) initChildWidget() {
-	if i.Child != nil {
-		return
+func (i *MenuItem) updateChildPosition() {
+	itemSize := i.Size()
+	cp := fyne.NewPos(itemSize.Width, -theme.Padding())
+	d := fyne.CurrentApp().Driver()
+	c := d.CanvasForObject(i)
+	if c != nil {
+		absPos := d.AbsolutePositionForObject(i)
+		childSize := i.Child.Size()
+		if absPos.X+itemSize.Width+childSize.Width > c.Size().Width {
+			if absPos.X-childSize.Width >= 0 {
+				cp.X = -childSize.Width
+			} else {
+				cp.X = c.Size().Width - absPos.X - childSize.Width
+			}
+		}
+		if absPos.Y+childSize.Height-theme.Padding() > c.Size().Height {
+			cp.Y = c.Size().Height - absPos.Y - childSize.Height
+		}
 	}
-	i.Child = NewMenu(i.Item.ChildMenu)
-	i.Child.Hide()
-	i.Child.DismissAction = func() { i.Parent.dismiss() }
+	i.Child.Move(cp)
 }
 
 type menuItemRenderer struct {
 	BaseRenderer
-	i    *MenuItem
-	icon *canvas.Image
-	text *canvas.Text
+	i       *MenuItem
+	icon    *canvas.Image
+	minSize fyne.Size
+	pad     fyne.Size
+	msPad   fyne.Size
```

Comment:
Hm … yes. I built this optimisation when I had a busy loop because of another issue (that I resolved). When debugging this I found that there were a lot of `MinSize` calls which led to the same size … so I put in the cache.

WDYT, should I remove it again?
Or should I just choose better names?
Something like `msPad` -> `lastItemPadding` and `tPad` -> `lastThemePadding`?

Reply:
That is tough I think your replacement solution works well.